### PR TITLE
feat: add deltalake sql crate

### PR DESCRIPTION
**Pull request by roeap**
# Description

This is an fairly early draft to create logical plans from sql using the datafusion abstractions. Adopted the patterns over there quite closely since the ultimate goal would be to ask the datafusion community if they would accept these changes within the datafusion sql crate ...

# Related Issue(s)

# Documentation

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -1,5 +1,5 @@
 [workspace]
-members = ["delta-inspect", "rust", "python"]
+members = ["delta-inspect", "rust", "python", "deltalake/*"]
```
Comment (rtyler) [inline Cargo.toml:2]: Interesting syntax :thinking:

I was actually refactoring the `deltalake` meta crate and `deltalake-core` crates over the weekend end got the impression that such wizardry was not possible to nest crates like this :thinking:

File: deltalake/sql/Cargo.toml [hunk 2]
```diff
@@ -0,0 +1,14 @@
+[package]
+name = "deltalake-sql"
+version = "0.1.0"
+edition = "2021"
```
Comment (rtyler) [inline deltalake/sql/Cargo.toml:4]: Before release this should pull version and license information from the workspace too

**Comment by rtyler** [comment]
@roeap I have done the conflict resolution to bring this up to date with `main` and pushed the branch. If you're up for it, I don't have a problem merging this in as it stands now and evolving more in `main` to a release

**Comment by roeap** [comment]
@rtyler - sounds good, as it does not do any harm on main, seems like the best way forward :).

**Review by rtyler — APPROVED** [review]

**Pull request merged**